ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -106,8 +106,10 @@ public NetworkClient(Selector selector, NetworkConfig networkConfig, NetworkMetr
         pendingRequests.add(new RequestMetadata(time.milliseconds(), requestInfo, clientNetworkRequestMetrics));
       }
       List<NetworkSend> sends = prepareSends(responseInfoList);
-      int connectionsInitiated = connectionTracker.replenishConnections(this::connect);
-      networkMetrics.connectionReplenished.inc(connectionsInitiated);
+      if (networkConfig.networkClientEnableConnectionReplenishment) {
+        int connectionsInitiated = connectionTracker.replenishConnections(this::connect);
+        networkMetrics.connectionReplenished.inc(connectionsInitiated);
```

Comment:
minor: do you need to update `networkMetrics.connectionReplenished` in method `warmUpConnections` as well?

Reply:
I think it is better to have this metric only apply to connections replenished in sendAndPoll